addon/utils/ref.js

```diff
@@ -17,6 +17,16 @@ export function setGlobalRef(value) {
   lastGlobalRef = value;
 }
 
+export function maybeCleanGlobalRef(value) {
+  if (lastGlobalRef === value) {
+    cleanGlobalRef();
+  }
+}
+
+export function cleanGlobalRef() {
+  lastGlobalRef = null;
+}
```

Comment:
This could be used in the application initializer but if everything is done properly, then I think we don't need both clean-ups (modifier destruction + app destruction).